Src/VimWpf/RelativeLineNumbers/LineNumberFormatTracker.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Globalization;
+using System.Linq;
+using System.Windows.Media;
+using System.Windows.Media.TextFormatting;
+
+using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Formatting;
+
+namespace Vim.UI.Wpf.RelativeLineNumbers
+{
+    internal class LineNumberFormatTracker : ILineFormatTracker
+    {
+        private readonly IWpfTextView _textView;
+        private readonly IFormattedLineSource _formattedLineSource;
+        private readonly IClassificationFormatMap _classificationFormatMap;
+        private readonly IClassificationTypeRegistryService _classificationTypeRegistry;
+
+        private TextFormattingRunProperties _formatting;
+
+        private TextFormatter _textFormatter;
+
+        private bool _formatChanged;
+
+        private bool _numbers;
+
+        private bool _relativeNumbers;
+
+        public Brush Background { get; private set; }
+
+        public double NumberWidth { get; private set; }
+
+        public bool Numbers
+        {
+            get => _numbers;
+            private set
+            {
+                if (_numbers != value)
+                {
+                    _numbers = value;
+                    OnVimNumbersFormatChanged();
+                }
+            }
+        }
+
+        public bool RelativeNumbers
+        {
+            get => _relativeNumbers;
+            private set
+            {
+                if (_relativeNumbers != value)
+                {
+                    _relativeNumbers = value;
+                    OnVimNumbersFormatChanged();
+                }
+            }
+        }
+
+        public LineNumberFormatTracker(
+            IWpfTextView textView,
+            IClassificationFormatMap classificationFormatMap,
+            IClassificationTypeRegistryService classificationTypeRegistry)
+        {
+            _textView = textView
+                ?? throw new ArgumentNullException(nameof(textView));
+
+            _classificationFormatMap = classificationFormatMap
+                ?? throw new ArgumentNullException(nameof(classificationFormatMap));
+
+            _classificationTypeRegistry = classificationTypeRegistry
+                ?? throw new ArgumentNullException(nameof(classificationTypeRegistry));
+
+            _formattedLineSource = textView.FormattedLineSource;
+
+            _classificationFormatMap.ClassificationFormatMappingChanged += (s, e) => UpdateFormat();
+
+            _textView.Options.OptionChanged += (s, e) => TryUpdateVimOption(e.OptionId);
+
+            UpdateFormat();
+        }
+
+        private void TryUpdateVimOption(string optionId)

```

Comment:
This type is essentially keeping a copy of the `numbers` and `relativenumbers` settings and having to do a bit of work to keep them in sync here. I think it would be simpler to just use `IVimGlobalSettings` directly here as the implementation. The values there will already be kept in sync via the `EditorToSettingsSynchronizer` type and it has an event that can be listened to for when the values change.

Reply:
Done. Don't we actually want to use LocalSettings here? So we can have different options for different file extensions/buffers?